src/components/Experience.ts

```diff
@@ -29,11 +30,86 @@ export class Experience<TGameStartr extends FullScreenPokemon> extends GeneralCo
+     * @returns True if the pokemon leveled up, false otherwise.
      */
-    public gainExperience(pokemon: IPokemon, experience: number): void {
+    public gainExperience(pokemon: IPokemon, experience: number): boolean {
         pokemon.experience += experience;
         if (pokemon.experience >= this.gameStarter.equations.experienceStarting(pokemon.title, pokemon.level + 1)) {
             this.levelup(pokemon);
+            return true;
         }
+        return false;
     }
+
+     /**
+      * Processes experience gain for one or more pokemon
+      *
+      * @param battleInfo   Information about the current battle.
+      * @param onComplete   Handler for when this is done.
+      *
+      */
+    public processBattleExperience(battleInfo: IBattleInfo, onComplete: () => void): void {
+        const experienceToGain =  this.gameStarter.equations.experienceGained(
+            battleInfo.teams.player, battleInfo.teams.opponent);
+
+        const experienceText: (string | string[])[][] = [[
+            battleInfo.teams.player.selectedActor.nickname,
+            ` gained ${experienceToGain} experience!`]];
+        this.gameStarter.menuGrapher.createMenu("GeneralText");
+        const levelUp = this.gainExperience(battleInfo.teams.player.selectedActor, experienceToGain);
+        if (levelUp) {
+            this.gameStarter.menuGrapher.addMenuDialog(
+                "GeneralText", experienceText,
+                () => this.processBattleLevelUp(battleInfo.teams.player.selectedActor, onComplete));
+        } else {
+            this.gameStarter.menuGrapher.addMenuDialog("GeneralText", experienceText, onComplete);
+        }
+        this.gameStarter.menuGrapher.setActiveMenu("GeneralText");
+    }
+
+     /**
+      * Processes level up for a given pokemon
+      *
+      * @param pokemon   Pokemon who is going to level up.
+      * @param onComplete   Handler for when this is done.
+      *
+      */
+    private processBattleLevelUp(pokemon: IPokemon, onComplete: () => void) {
+        this.gameStarter.menuGrapher.createMenu("GeneralText");
+        const experienceText: (string | string[])[][] = [[
+            pokemon.nickname,
+            ` grew to level ${pokemon.level}!`]];
+        if (this.canLearnMove(pokemon)) {
+            this.gameStarter.menuGrapher.addMenuDialog(
+                "GeneralText", experienceText,
+                () => this.learnBattleMove(pokemon, onComplete));
+        } else {
+            this.gameStarter.menuGrapher.addMenuDialog(
+                "GeneralText", experienceText, onComplete);
+        }
+        this.gameStarter.menuGrapher.setActiveMenu("GeneralText");
+    }
+
+     /**
+      * Determines whether a pokemon should or shouldn't learn a move
+      *
+      * @param pokemon   Pokemon who is going to going to learn a move.
+      * @returns True is pokemon can learn a move at the given level, false otherwise
+      *
+      */
+    private canLearnMove(pokemon: IPokemon): boolean {
```

Comment:
This function name+description should mention that it's at a specific level. There are functions elsewhere in code that also check whether Pokemon can learn a move.